### PR TITLE
Convert citizen and solicitor application submission to CaseTasks

### PR DESCRIPTION
**Pull request by mapingo**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/NFDIV-1042

### Change description ###
Convert citizen and solicitor application submission to use the same set of CaseTasks

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/divorce/citizen/event/CitizenAddPayment.java [hunk 1]
```diff
@@ -57,51 +49,48 @@ public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuild
...
+            return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+                .data(details.getData())
+                .state(Draft)
+                .build();
+        }
+
+        log.info("Validating case caseData CaseID: {}", caseId);
+        final List<String> submittedErrors = Submitted.validate(caseData);
+
+        if (!submittedErrors.isEmpty()) {
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/citizen/event/CitizenAddPayment.java:81]: this whole method is much better

File: src/main/java/uk/gov/hmcts/divorce/common/service/task/SendSubmissionNotifications.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        } else {
+            sendCitizenNotifications(caseData, caseId, state);
+        }
+
+        return caseDetails;
+    }
+
+    private void sendCitizenNotifications(final CaseData caseData, final Long caseId, final State state) {
+
+        if (Submitted.equals(state)) {
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/common/service/task/SendSubmissionNotifications.java:48]: I think this is Submitted, AwaitingDocuments or HWF

File: src/main/java/uk/gov/hmcts/divorce/common/service/task/SetDateSubmitted.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    @Autowired
+    private Clock clock;
+
+    @Override
+    public CaseDetails<CaseData, State> apply(final CaseDetails<CaseData, State> caseDetails) {
+
+        final CaseData caseData = caseDetails.getData();
+        final State state = caseDetails.getState();
+
+        if (Submitted.equals(state) || AwaitingDocuments.equals(state)) {
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/common/service/task/SetDateSubmitted.java:30]: I think we can add HWF here

**Review by linusnorton — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/divorce/common/service/task/SendSubmissionNotifications.java (hunk 2 above)
Comment (mapingo) [reply src/main/java/uk/gov/hmcts/divorce/common/service/task/SendSubmissionNotifications.java]: Updated in https://github.com/hmcts/nfdiv-case-api/pull/571/commits/3f45324265a75c89a817a5cbb3e14219d7414d19

File: src/main/java/uk/gov/hmcts/divorce/common/service/task/SetDateSubmitted.java (hunk 3 above)
Comment (mapingo) [reply src/main/java/uk/gov/hmcts/divorce/common/service/task/SetDateSubmitted.java]: Updated in https://github.com/hmcts/nfdiv-case-api/pull/571/commits/3f45324265a75c89a817a5cbb3e14219d7414d19

**Pull request merged**